diff --git a/includes/context.inc b/includes/context.inc
index 93be748..ade75c8 100644
--- a/includes/context.inc
+++ b/includes/context.inc
@@ -211,7 +211,8 @@ class ctools_context_optional extends ctools_context_required {
     $context = new ctools_context('any');
     $context->title      = t('No context');
     $context->identifier = t('No context');
-    $contexts = array_merge(array('empty' => $context), $contexts);
+    //$contexts = array_merge(array('empty' => $context), $contexts);
+    $contexts['empty'] = $context;
   }
 
   function filter($contexts) {
