diff --git a/pp_crop.module b/pp_crop.module
index a86851f..abb031c 100644
--- a/pp_crop.module
+++ b/pp_crop.module
@@ -111,6 +111,10 @@ function pp_crop_element_process($element, $form_state, $form) {
 
   _epsacrop_include_header_html();
   $styles = _epsacrop_load_styles_by_instance($element['#entity_type'], $element['#field_name'], $element['#bundle']);
+
+  // Allow other modules to alter list of styles displayed in the popup.
+  drupal_alter('epsacrop_load_styles', $styles, $element['#entity_type'], $element['#field_name'], $element['#bundle']);
+
   if (count($styles) == 0) {
     return $element;
   }
@@ -321,8 +325,8 @@ function pp_crop_ajax_manage_thumbnails($settings, $fid) {
 
   $file = file_load($fid);
 
-  $field_settings = _epsacrop_load_styles_by_instance($theme_variables['entity_type'], $theme_variables['field_name'], $theme_variables['bundle']);
-  foreach ($field_settings['styles'] as $image_style) {
+  $styles = $theme_variables['styles'];
+  foreach ($styles as $image_style) {
     $thumbnail_options[$image_style] = theme('image_style', array('style_name' => $image_style, 'path' => $file->uri, 'attributes' => array('class' => 'pp-crop-thumbnail')));
   }
 
