diff --git a/netx/js/netx.cfb.js b/netx/js/netx.cfb.js
index 0898e3b..a52693b 100644
--- a/netx/js/netx.cfb.js
+++ b/netx/js/netx.cfb.js
@@ -305,17 +305,87 @@
                     pager.empty();
                     if (scannedPagers.length) {
                         scannedPagers.forEach(function (f) {
-                            var $pager_list = $('<ul class="netx-pager"></ul>')
-                            for (var i = 0; i < f.pages; i++) {
-                                var $pager_item = $('<li class="netx-pager-item">' +
-                                  '<a href="' + currentPath + '" data-page="' + i + '">' + (i + 1) + '</a>' +
-                                  '</li>');
-                                if (i == parseInt(f.current)) {
-                                    $pager_item.addClass('current');
+                            // If pages more one add pagination.
+                            if (f.pages > 1) {
+                                var $pagerList = $('<ul class="netx-pager"></ul>'),
+                                    $pagerMiddle = 5,
+                                    // Current is the page we are currently paged to.
+                                    $pagerCurrent = parseInt(f.current),
+                                    // First is the first page listed by this pager piece (re quantity).
+                                    $pagerFirst = $pagerCurrent - $pagerMiddle,
+                                    // Last is the last page listed by this pager piece (re quantity).
+                                    $pagerLast = $pagerCurrent + $pagerMiddle,
+                                    // Max is the maximum page number.
+                                    $pagerMax = f.pages - 1,
+                                    $pagerItem = '';
+
+                                // Prepare item offset for generation loop.
+                                var $item = $pagerFirst;
+                                // Adjust "center" if at end of query.
+                                if ($pagerLast > $pagerMax) {
+                                    $item = $item + ($pagerMax - $pagerLast);
+                                    $pagerLast = $pagerMax;
                                 }
-                                $pager_item.appendTo($pager_list);
+                                // Adjust "center" if at start of query.
+                                if ($item <= 0) {
+                                    $pagerLast = $pagerLast + (1 - $item);
+                                    $item = 1;
+                                }
+                                // Add Previous page link or text.
+                                if ($pagerCurrent != 0) {
+                                    $pagerItem = $('<li class="netx-pager-item">' +
+                                        '<a href="' + currentPath + '" data-page="' + ($pagerCurrent - 1) + '">' + Drupal.t('Previous page') + '</a>' + '</li>');
+                                    $pagerItem.appendTo($pagerList);
+                                }
+                                else {
+                                    $pagerItem = $('<li class="netx-pager-item">' +
+                                        '<span>' + Drupal.t('Previous page') + '</span>' + '</li>');
+                                    $pagerItem.appendTo($pagerList);
+                                }
+                                // Add first item.
+                                $pagerItem = $('<li class="netx-pager-item">' +
+                                    '<a href="' + currentPath + '" data-page="0">1</a>' + '</li>');
+                                $pagerItem.appendTo($pagerList);
+
+                                if ($item > 1) {
+                                    $pagerItem = $('<li class="netx-pager-item">' +
+                                        '<span>...</span>' + '</li>');
+                                    $pagerItem.appendTo($pagerList);
+                                }
+                                // Generate page list.
+                                for (var i = $item; i < $pagerLast; i++) {
+                                    $pagerItem = $('<li class="netx-pager-item">' +
+                                        '<a href="' + currentPath + '" data-page="' + i + '">' + (i + 1) + '</a>' +
+                                        '</li>');
+                                    if (i == $pagerCurrent) {
+                                        // Add specific class for current item.
+                                        $pagerItem.addClass('current');
+                                    }
+                                    $pagerItem.appendTo($pagerList);
+                                }
+
+                                if ($item < $pagerMax) {
+                                    $pagerItem = $('<li class="netx-pager-item">' +
+                                        '<span>...</span>' + '</li>');
+                                    $pagerItem.appendTo($pagerList);
+                                }
+                                // Add last item.
+                                $pagerItem = $('<li class="netx-pager-item">' +
+                                    '<a href="' + currentPath + '" data-page="' + $pagerMax + '">' + ($pagerMax + 1) + '</a>' + '</li>');
+                                $pagerItem.appendTo($pagerList);
+                                // Add Next page link or text.
+                                if ($pagerCurrent != $pagerMax) {
+                                    $pagerItem = $('<li class="netx-pager-item">' +
+                                        '<a href="' + currentPath + '" data-page="' + ($pagerCurrent + 1) + '">' + Drupal.t('Next page') + '</a>' + '</li>');
+                                    $pagerItem.appendTo($pagerList);
+                                }
+                                else {
+                                    $pagerItem = $('<li class="netx-pager-item">' +
+                                        '<span>' + Drupal.t('Next page') + '</span>' + '</li>');
+                                    $pagerItem.appendTo($pagerList);
+                                }
+                                $pagerList.appendTo(pager);
                             }
-                            $pager_list.appendTo(pager);
                         });
                     }
 
