diff --git a/drupal/sites/all/modules/contrib/bean_wysiwyg/plugins/bean/bean.js b/drupal/sites/all/modules/contrib/bean_wysiwyg/plugins/bean/bean.js
index c9e782e..2f0b035 100644
--- a/drupal/sites/all/modules/contrib/bean_wysiwyg/plugins/bean/bean.js
+++ b/drupal/sites/all/modules/contrib/bean_wysiwyg/plugins/bean/bean.js
@@ -183,7 +183,7 @@
    * Insert HTML to the WYSIWYG when block has been created. Set the macro.
    */
     insert: function (block) {
-      var markup = '<p>&nbsp;</p>' + bean_wysiwyg_create_markup(block) + '<p>&nbsp;</p>';
+      var markup = bean_wysiwyg_create_markup(block);
       macro = bean_wysiwyg_create_macro(markup);
 
       // Insert placeholder markup into wysiwyg.
