diff --git a/docroot/sites/all/modules/contrib/entitycache/entitycache.module b/docroot/sites/all/modules/contrib/entitycache/entitycache.module
index 66366e0..ba297d8 100644
--- a/docroot/sites/all/modules/contrib/entitycache/entitycache.module
+++ b/docroot/sites/all/modules/contrib/entitycache/entitycache.module
@@ -212,3 +212,22 @@ function entitycache_user_logout($account) {
 function entitycache_user_login(&$edit, $account) {
   cache_clear_all($account->uid, 'cache_entity_user');
 }
+
+if (!class_exists('EntityCacheUserController')) {
+  require_once __DIR__ . '/includes/entitycache.user.inc';
+}
+if (!class_exists('EntityCacheTaxonomyTermController')) {
+  require_once __DIR__ . '/includes/entitycache.taxonomy.inc';
+}
+if (!class_exists('EntityCacheTaxonomyVocabularyController')) {
+  require_once __DIR__ . '/includes/entitycache.taxonomy.inc';
+}
+if (!class_exists('EntityCacheControllerHelper')) {
+  require_once __DIR__ . '/includes/entitycache.entitycachecontrollerhelper.inc';
+}
+if (!class_exists('EntityCacheDefaultEntityController')) {
+  require_once __DIR__ . '/includes/entitycache.defaultentitycontroller.inc';
+}
+if (!class_exists('EntityCacheNodeController')) {
+  require_once __DIR__ . '/includes/entitycache.node.inc';
+}
