diff --git a/i18n/i18n_node/i18n_node.pages.inc b/i18n/i18n_node/i18n_node.pages.inc
index 5b4e62e..f7b22c1 100644
--- a/docroot/sites/all/modules/patched/i18n/i18n_node/i18n_node.pages.inc
+++ b/docroot/sites/all/modules/patched/i18n/i18n_node/i18n_node.pages.inc
@@ -125,6 +125,9 @@ function i18n_node_translation_link($text, $path, $langcode, $options = array())
     $options += $links->links[$langcode];
     $path = $links->links[$langcode]['href'];
   }
+  $languages = language_list();
+  $language = $languages[$langcode];
+  $options += array('language' => $language);
   return l($text, $path, $options);
 }
