diff --git a/docroot/sites/all/modules/contrib/i18n/i18n_node/i18n_node.pages.inc b/docroot/sites/all/modules/contrib/i18n/i18n_node/i18n_node.pages.inc
index 4e13ba6..a104ff1 100644
--- a/docroot/sites/all/modules/contrib/i18n/i18n_node/i18n_node.pages.inc
+++ b/docroot/sites/all/modules/contrib/i18n/i18n_node/i18n_node.pages.inc
@@ -51,6 +51,11 @@ function i18n_node_translation_overview($node) {
     $translations = array($node->language => $node);
   }
 
+  $default_language = language_default();
+  if ($node->language == LANGUAGE_NONE) {
+    $translations[$default_language->language] = &$translations[$node->language];
+  }
+
   $type = variable_get('translation_language_type', LANGUAGE_TYPE_INTERFACE);
   $header = array(t('Language'), t('Title'), t('Status'), t('Operations'));
 
