diff --git a/docroot/sites/all/modules/contrib/imagecache_token/imagecache_token.module b/docroot/sites/all/modules/contrib/imagecache_token/imagecache_token.module
index c083ec8..93087c2 100644
--- a/docroot/sites/all/modules/contrib/imagecache_token/imagecache_token.module
+++ b/docroot/sites/all/modules/contrib/imagecache_token/imagecache_token.module
@@ -227,7 +227,8 @@ function imagecache_token_tokens($type, $tokens, array $data = array(), array $o
             }
           }
           else {
-            $output[] = image_style_url($token, $field['uri']);
+            $uri = !empty($field['file']->uri) ? $field['file']->uri : $field['uri'];
+            $output[] = image_style_url($token, $uri);
           }
         }
 
